msnoise/s01scan_archive.py

```diff
@@ -50,210 +52,519 @@
+    :param basename: the basename of the miniseed file to read data from
+    :param stream: unfiltered obspy.core.stream.Stream object
+    :param id_: the id of the stream to read
+    :param startdate: the startdate configuration value
+    :param enddate: the enddate configuration value
+    :param goal_sampling_rate: the sampling rate
+    """
+    net, sta, loc, chan = id_.split('.')
+    data = stream.select(network=net, station=sta, location=loc, channel=chan)
+    if data[-1].stats.endtime.date < startdate:
+        #logger.debug('ignoring %s: before start date!' % id)
+        return 0
+    if data[0].stats.starttime.date > enddate:
+        #logger.debug('ignoring %s: after end date!' % id)
+        return 0
+    if data[0].stats.sampling_rate < goal_sampling_rate:
+        #logger.debug('ignoring %s: sampling rate smaller than '
+        #              'CC sampling rate' % id)
+        return 0
+    return update_availability(db, folder, basename, data)
+
+
+def scan_data_files(db, folder, files, startdate, enddate, goal_sampling_rate,
+        logger):
+    """
+    Processes a list of files from a folder, and update the data availability
+    table in the database whenever their data matches our dates and sampling
+    rate parameters.
+
+    :param db: the sqlalchemy session object.
+    :param folder: the directory where lies the miniseed files.
+    :param files: the list of files of the folder to read.
+    :param stream: unfiltered obspy.core.stream.Stream object.
+    :param id_: the id of the stream to read.
+    :param startdate: the startdate configuration value.
+    :param enddate: the enddate configuration value.
+    :param goal_sampling_rate: the sampling rate.
+    :paran logger: the logger instance to use for logging.
+    """
     added = 0
     modified = 0
     unchanged = 0
-    for file in files:
-        if init:
-            file = os.path.join(folder, file)
+    for basename in files:
+        pathname = os.path.join(folder, basename)
+        #logger.debug('reading file %s' % pathname)
         try:
-            name = os.path.split(file)[1]
-            st = read(file, headonly=True)
-            unique_ids = np.unique([tr.id for tr in st])
-            for id in unique_ids:
-                net, sta, loc, chan = id.split(".")
-                data = st.select(network=net, station=sta,
-                                 location=loc, channel=chan)
-                if data[-1].stats.endtime.date < startdate:
-                    logging.debug(
-                        '%s: Before Start-Date!' % (id))
-                elif data[0].stats.starttime.date > enddate:
-                    logging.debug('%s: After End-Date!' % (id))
-                elif data[0].stats.sampling_rate < goal_sampling_rate:
-                    logging.debug("%s: Sampling rate smaller than CC sampling"
-                                 " rate" % id)
+            stream = obspy.core.read(pathname, headonly=True, format='MSEED')
```

Comment:
don't force MSEED here, it can be different.
maybe add a ``default`` parameter in the config when users know what their format is, so that it prevents ObsPy from scanning all possible formats.

Reply:
This is hopefully resolved by df23117 that adds an `archive_format` configuration parameter to force the format (if empty or unknown, obspy will use its auto-detection capability).